packages/clay-data-provider/src/useResource.ts

```
@@ -0,0 +1,228 @@
+	},
+	fetchRetry = {},
+	storageMaxSize = 20,
+	variables = null,
+}: IResource) => {
+	const [resource, setResource] = useState<any>(null);
+
+	let networkStatus = useRef<NetworkStatus>(NetworkStatus.Unused).current;
+
+	let pollingIntervalId = useRef<null | NodeJS.Timeout>(null).current;
+
+	let retryDelayIntervalId = useRef<null | NodeJS.Timeout>(null).current;
+
+	// A flag to identify if the first rendering happened to avoid
+	// two requests.
+	const firstRenderRef = useRef<boolean>(true);
+
+	const cache = useCache(
+		fetchPolicy,
+		storage,
+		storageMaxSize,
+		link,
+		variables
+	);
+
+	const debouncedVariablesChange = useDebounce(variables, fetchDelay);
+
+	const setNetworkStatus = (status: NetworkStatus) => {
+		networkStatus = status;
+		onNetworkStatusChange(status);
+	};
+
+	const handleRefetch = () => {
+		doFetch();
+		setNetworkStatus(NetworkStatus.Refetch);
+	};
+
+	const cleanRetry = () => {
+		if (retryDelayIntervalId) {
+			clearInterval(retryDelayIntervalId);
+		}
+	};
+
+	const getRetryDelay = (retryAttempts: number) => {
+		const {delay = {}} = fetchRetry;
+		const {initial = 300, jitter = true} = delay;
+
+		const baseDelay = jitter ? initial : initial / 2;
+
+		let value = baseDelay * 2 ** retryAttempts;
+
+		if (jitter) {
+			value = Math.random() * value;
+		}
+
+		return value;
+	};
+
+	const handleFetchRetry = (err: any, retryAttempts: number) => {
+		const {attempts = 5} = fetchRetry;
+
+		cleanRetry();
+
+		if (attempts > 0 && retryAttempts < attempts) {
+			const delay = getRetryDelay(retryAttempts);
+
+			// eslint-disable-next-line no-console
+			console.warn(
+				`DataProvider: Trying ${retryAttempts +
+					1} of ${attempts} will happen in ${delay}ms`
+			);
+
+			retryDelayIntervalId = setInterval(() => {
+				doFetch(retryAttempts + 1);
+			}, delay);
+		} else {
+			setNetworkStatus(NetworkStatus.Error);
+			// eslint-disable-next-line no-console
+			console.error('DataProvider: Error making the requisition', err);
```

Should we throw an error here instead or is this more of a warning? If its just a warning, lets use the `warning` package.

---

I think we can use the `warning` package here, we may have some confusion with this package, in fact it throws an `error`.